tiledb/sm/object/object.cc

```diff
@@ -77,32 +77,33 @@ bool is_array(ContextResources& resources, const URI& uri) {
   }
 }
 
-Status is_group(ContextResources& resources, const URI& uri, bool* is_group) {
+bool is_group(ContextResources& resources, const URI& uri) {
   // Handle remote array
   if (uri.is_tiledb()) {
     auto&& [st, exists] =
         resources.rest_client()->check_group_exists_from_rest(uri);
     throw_if_not_ok(st);
-    *is_group = *exists;
+    assert(exists.has_value());
```

Comment:
Avoid `assert` in new code. Either it's important enough to throw or the check can be omitted. In this case it's sufficient to rely on the postcondition that if the status is ok then the optional is not null. If there's a violation (and thus it's 'nullopt`), `value` will throw. Thus a separate check and throw doesn't add much at all.